### DISCUSSION TITLE
Development environment

### DISCUSSION THREAD

**Discussion by Sylvain-Caron** [OCA/odoo-community.org#113] [Q&A]
Hi, 

My name is Sylvain.

I would like to know about creating an Odoo local environment, because I want to be more efficient in the development of the modules, than on Odoo.sh.
I can already launch my server with Odoo, Odoo enterprise and my custom_addons.

However I have other problems like updating the code live, I have to constantly restart the server. I saw that there were default parameters for development like: --dev=reload using the "watchdog" lib, except that after 2 3 refresh the server shuts down.

I have already used Odoo.sh and its editor, and Python recompiles well with each change, so this is doable

Do you have an installation guide for local development?
The development by Docker is perhaps simpler?

Some additional information:

I am on windows
VScode, and I'm using vscode launch.js configs for Odoo server

Thanks in advance

**Comment by marcelsavegnago** [comment]
https://github.com/Tecnativa/doodba
https://github.com/akretion/docky
